src/pages/Lending/hooks/useAllLendingPositionsData.tsx

```diff
@@ -5,57 +5,68 @@ import { bn } from 'lib/bignumber/bignumber'
 import type { Borrower } from 'state/slices/opportunitiesSlice/resolvers/thorchainLending/types'
 import { getThorchainLendingPosition } from 'state/slices/opportunitiesSlice/resolvers/thorchainLending/utils'
 import { fromThorBaseUnit } from 'state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils'
-import { selectAccountIdsByAssetId, selectUsdRateByAssetId } from 'state/slices/selectors'
+import { selectAccountIdsByAssetId, selectMarketDataById } from 'state/slices/selectors'
 import { store } from 'state/store'
 
 import { useLendingSupportedAssets } from './useLendingSupportedAssets'
 
-export const useAllLendingPositionsData = () => {
+type UseAllLendingPositionsDataProps = {
+  assetId?: AssetId
+}
+
+export const useAllLendingPositionsData = ({ assetId }: UseAllLendingPositionsDataProps = {}) => {
   const { data: lendingSupportedAssets } = useLendingSupportedAssets()
 
   const accounts = useMemo(
     () =>
       (lendingSupportedAssets ?? [])
         .map(asset => {
-          const assetId = asset.assetId
-          const _accountIds = selectAccountIdsByAssetId(store.getState(), { assetId })
-          return _accountIds.map(accountId => ({ accountId, assetId }))
+          if (assetId && assetId !== asset.assetId) return []
+          const _accountIds = selectAccountIdsByAssetId(store.getState(), {
+            assetId: asset.assetId,
+          })
```

Comment:
i wonder what the performance implications are of selecting inside a loop like this versus selecting the whole outside and keying here?

Reply:
Conceptually, absolutely agree here - in that case this should be most likely fine as there are only two collaterals so far, but this would be a potential performance bottleneck as there are more collaterals, or we end up consuming `findAccountsByAssetId` / `selectAccountIdsByAssetId` in more places.

Tried to refactor this so that we return the whole object of `PartialRecord<AsssetId, AccountId[]>` and key by `AssetId`, however, we can't do this since an `AccountId` doesn't contain enough data to get an `AssetId` to use as a key if we want to return all and don't use an `AssetId` filter 😢 

`account.assetIds` *would* contain such data, though we end up in a state where we have to iterate over all account `AssetId`s to be able to index by `AssetId`, which now makes things more iteraty and minimizes the performance benefits.

Reply:
Ah makes sense, appreciate your investigation ser. Lets roll with this solution